src/main/java/com/minecolonies/coremod/util/ItemStackUtils.java

```diff
@@ -0,0 +1,441 @@
+
+    /**
+     * Verifies if there is one tool with an acceptable level
+     * in a worker's inventory.
+     *
+     * @param stack        the stack to test.
+     * @param toolType     the type of tool needed
+     * @param minimalLevel the minimum level for the tool to find.
+     * @param maximumLevel the maximum level for the tool to find.
+     * @return true if tool is acceptable
+     */
+    public static boolean hasToolLevel(@Nullable final ItemStack stack, final IToolType toolType, final int minimalLevel, final int maximumLevel)
+    {
+        if (isEmpty(stack))
+        {
+            return false;
+        }
+
+        final int level = getMiningLevel(stack, toolType);
+        return isTool(stack, toolType) && verifyToolLevel(stack, level, minimalLevel, maximumLevel);
+    }
+
+    /**
+     * Wrapper method to check if a stack is empty.
+     * Used for easy updating to 1.11.
+     *
+     * @param stack The stack to check.
+     * @return True when the stack is empty, false when not.
+     */
+    public static Boolean isEmpty(@Nullable final ItemStack stack)
+    {
+        return stack == EMPTY || stack.getItem() == null || stack.stackSize <= 0;
+    }
+
+    /**
+     * Checks if this ItemStack can be used as a Tool of type.
+     *
+     * @param itemStack Item to check.
+     * @param toolType  Type of the tool.
+     * @return true if item can be used, otherwise false.
+     */
+    public static boolean isTool(@Nullable final ItemStack itemStack, final IToolType toolType)
+    {
+        if (isEmpty(itemStack))
+        {
+            return false;
+        }
+
+        boolean isATool = false;
+        if (ToolType.AXE.equals(toolType) || ToolType.SHOVEL.equals(toolType) || ToolType.PICKAXE.equals(toolType))
+        {
+            isATool = getMiningLevel(itemStack, toolType) >= 0;
+        }
+        else if (ToolType.HOE.equals(toolType))
+        {
+            isATool = itemStack.getItem() instanceof ItemHoe;
+        }
+        else if (ToolType.BOW.equals(toolType))
+        {
+            isATool = itemStack.getItem() instanceof ItemBow;
+        }
+        else if (ToolType.SWORD.equals(toolType))
+        {
+            isATool = itemStack.getItem() instanceof ItemSword;
+        }
+        else if (ToolType.FISHINGROD.equals(toolType))
+        {
+            isATool = itemStack.getItem() instanceof ItemFishingRod;
+        }
+        return isATool;
+    }
+
+
+    /**
+     * Calculate the mining level an item has as a tool of certain type.
+     * Suppressing Sonar Rule squid:S1142
+     * This rule does "Methods should not have too many return statements"
+     * I could limit the number of return by using a variable and only one return
+     * but it will not make the code clearer
+     *
+     * @param stack    the stack to test.
+     * @param toolType the tool category.
+     * @return integer value for mining level &gt;= 0 is okay.
+     */
+    @SuppressWarnings({"deprecation", "squid:S1142"})
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Suppressing the 'squid:S1142' warning is not allowed [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1309)